sha2/Cargo.toml

```diff
@@ -15,22 +15,22 @@ keywords = ["crypto", "sha2", "hash", "digest"]
 categories = ["cryptography", "no-std"]
 
 [dependencies]
-digest = "0.10.2"
+digest = "0.10.3"
 cfg-if = "1.0"
 sha2-asm = { version = "0.6.1", optional = true }
 
 [target.'cfg(any(target_arch = "aarch64", target_arch = "x86_64", target_arch = "x86"))'.dependencies]
 cpufeatures = "0.2"
 
 [dev-dependencies]
-digest = { version = "0.10.2", features = ["dev"] }
-hex-literal = "0.2"
+digest = { version = "0.10.3", features = ["dev"] }
+hex-literal = "0.2.2"
 
 [features]
 default = ["std"]
 std = ["digest/std"]
-asm = ["sha2-asm"]
-compress = [] # Expose compress function
+asm = ["sha2-asm"] # WARNING: this feature SHOULD NOT be enabled by library crates
```

Comment:
Can you clarify why was this warning added?

Reply:
Because if a crate somewhere deep in your dependency tree enables it, it could become virtually impossible to disable it. Considering that the assembly crates may fail to build on some targets (e.g. see https://github.com/RustCrypto/asm-hashes/issues/34), it can be quite unpleasant for users downstream.

In future we may migrate to the configuration flags approach which we now use in some cipher crates, e.g. see the [relevant `aes` docs](https://docs.rs/aes/latest/aes/#configuration-flags).

Reply:
Link to the issue is exactly what I was looking for, thanks!